onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1993,15 +1993,20 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<FusedNodeAnd
     // Otherwise engine will be handled at inference time.
     std::unique_ptr<nvinfer1::ICudaEngine> trt_engine;
     std::unique_ptr<nvinfer1::IExecutionContext> trt_context;
+
+    // Name the engine cache based on GPU compute capacity to prevent user from loading an incompatible cache
```

Comment:
"partially" prevent

Note: for example engine cache generated for RTX 3090 (24GB memory) might not run in RTX 3050 (6GB memory?) even though they have same compute capactiy.

Reply:
Thanks for pointing that out! Comments have been updated